pkg/initializr/initializr.go

```diff
@@ -0,0 +1,166 @@
+package initializr
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// InitializrRequest contains options supported to generate a project using https://github.com/spring-io/initializr
+type InitializrRequest struct {
+	Dependencies string
+	GroupID      string
+	ArtifactID   string
+	Path         string
+}
+
+// InitializrResponse contains the response from a project generation request
+type InitializrResponse struct {
+	ContentType string
+	Contents    []byte
+	Filename    string
+}
+
+// CreateNewProject creates a new project using start.spring.io
+func CreateNewProject(request InitializrRequest) error {
+	resp, err := generate(request)
+	if err != nil {
+		return err
+	}
+	if request.Path == "" {
+		workingDir, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		return unpack(resp.Contents, workingDir)
+	}
+	return unpack(resp.Contents, request.Path)
+
+}
+
+func generate(request InitializrRequest) (InitializrResponse, error) {
+
+	u, err := url.Parse("https://start.spring.io/starter.zip")
+	if err != nil {
+		return InitializrResponse{}, err
+	}
+
+	q := u.Query()
+	q.Set("dependencies", request.Dependencies)
+	q.Set("groupId", request.GroupID)
+	q.Set("artifactId", request.ArtifactID)
+	q.Set("type", "maven-project")
+	u.RawQuery = q.Encode()
+
+	log.Debug("Initializr encoded URL: ", u.String())
+
+	// default timeout is infinite...
+	var httpClient = &http.Client{
```

Comment:
We should consider a pattern like [1] to make the HTTP Client more testable in an isolated way

[1] https://github.com/pivotal-cf/om/blob/master/network/trace_client.go

Reply:
Created https://github.com/cloudlift/lift/issues/13